compiler/src/main/kotlin/edu/cornell/cs/apl/viaduct/syntax/ProtocolPort.kt

```diff
@@ -0,0 +1,21 @@
+package edu.cornell.cs.apl.viaduct.syntax
+
+typealias PortId = String
+
+interface ProtocolPort {
+    val protocol: Protocol
+    val host: Host
+    val id: PortId
+}
+
+data class InputPort(
+    override val protocol: Protocol,
```

Comment:
"Make invalid states impossible". Isn't putting `protocol` here asking for mismatched data? If I get a port from `Replication`, I expect `protocol` to be `Replication`. What if it isn't? I think we should remove this, and ask people to remember the port.

Reply:
i thought you weren't working on the internal compiler representations anymore :)